test/gtest/groupnorm.hpp

```diff
@@ -0,0 +1,295 @@
+            {32, 1, 14, 14, 14, 1, 1e-5, MIOPEN_WEIGHT_BIAS},
+            {32, 32, 14, 14, 14, 4, 1e-5, MIOPEN_WEIGHT_BIAS},
+            {32, 32, 12, 12, 12, 4, 1e-5, MIOPEN_WEIGHT_BIAS},
+            {32, 32, 6, 6, 6, 4, 1e-5, MIOPEN_WEIGHT_BIAS},
+            {256, 1, 32, 32, 32, 1, 1e-5, MIOPEN_WEIGHT_BIAS},
+            {256, 32, 14, 14, 14, 4, 1e-5, MIOPEN_WEIGHT_BIAS},
+            {256, 32, 12, 12, 12, 4, 1e-5, MIOPEN_WEIGHT_BIAS},
+            {256, 32, 6, 6, 6, 4, 1e-5, MIOPEN_WEIGHT_BIAS},
+            {512, 1, 32, 32, 32, 1, 1e-5, MIOPEN_WEIGHT_BIAS},
+            {512, 32, 14, 14, 14, 4, 1e-5, MIOPEN_WEIGHT_BIAS},
+            {512, 32, 12, 12, 12, 4, 1e-5, MIOPEN_WEIGHT_BIAS},
+            {512, 32, 6, 6, 6, 4, 1e-5, MIOPEN_WEIGHT_BIAS},
+            {32, 2, 32, 57, 125, 2, 1e-5, MIOPEN_WEIGHT_BIAS},
+            {32, 32, 14, 25, 59, 4, 1e-5, MIOPEN_WEIGHT_BIAS},
+            {32, 32, 6, 10, 27, 4, 1e-5, MIOPEN_WEIGHT_BIAS},
+            {32, 32, 4, 6, 11, 4, 1e-5, MIOPEN_WEIGHT_BIAS},
+            {32, 32, 2, 2, 3, 4, 1e-5, MIOPEN_WEIGHT_BIAS},
+            {32, 32, 32, 28, 62, 4, 1e-5, MIOPEN_WEIGHT_BIAS},
+            {32, 32, 14, 12, 29, 4, 1e-5, MIOPEN_WEIGHT_BIAS},
+            {32, 32, 6, 4, 12, 4, 1e-5, MIOPEN_WEIGHT_BIAS},
+            {32, 32, 4, 2, 2, 4, 1e-5, MIOPEN_WEIGHT_BIAS},
+            {16, 32, 6, 50, 50, 4, 1e-5, MIOPEN_WEIGHT_BIAS},
+            // {1, 3, 8, 240, 320, 1, 1e-5, MIOPEN_WEIGHT_BIAS},
+            // {1, 3, 16, 240, 320, 1, 1e-5, MIOPEN_WEIGHT_BIAS},
+            // {1, 3, 8, 128, 171, 1, 1e-5, MIOPEN_WEIGHT_BIAS},
+            // {1, 3, 16, 128, 171, 1, 1e-5, MIOPEN_WEIGHT_BIAS},
+            // {1, 3, 8, 112, 112, 1, 1e-5, MIOPEN_WEIGHT_BIAS},
+            // {1, 3, 16, 112, 112, 1, 1e-5, MIOPEN_WEIGHT_BIAS},
+            {32, 4, 0, 4, 256, 1, 1e-5, MIOPEN_ELEMENTWISE_AFFINE},
+            {64, 4, 0, 4, 256, 1, 1e-5, MIOPEN_ELEMENTWISE_AFFINE},
+            {32, 4, 0, 4, 256, 1, 1e-5, MIOPEN_WEIGHT_BIAS},
+            {64, 4, 0, 4, 256, 1, 1e-5, MIOPEN_WEIGHT_BIAS},
+            {32, 1, 0, 0, 256, 1, 1e-5, MIOPEN_ELEMENTWISE_AFFINE},
+            {64, 1, 0, 0, 256, 1, 1e-5, MIOPEN_ELEMENTWISE_AFFINE},
+            {32, 1, 0, 0, 256, 1, 1e-5, MIOPEN_WEIGHT_BIAS},
+            {64, 1, 0, 0, 256, 1, 1e-5, MIOPEN_WEIGHT_BIAS}};
+}
+
+inline int32_t SetTensorLayout(miopen::TensorDescriptor& desc)
+{
+    std::vector<std::size_t> lens = desc.GetLengths();
+    std::vector<int32_t> int32_t_lens(lens.begin(), lens.end());
+
+    // set the strides for the tensor
+    return SetTensorNd(&desc, int32_t_lens, desc.GetType());
+}
+
+template <typename T = float>
+struct GroupNormTest : public ::testing::TestWithParam<GroupNormTestCase>
+{
+protected:
+    void SetUp() override
+    {
+        auto&& handle    = get_handle();
+        groupnorm_config = GetParam();
+        std::mt19937 gen(0);
+        std::uniform_real_distribution<> d{-3, 3};
+        auto gen_value = [&](auto...) { return d(gen); };
```

Comment:
Please use random generators from `driver/random.hpp test/random.hpp`
Check this comment https://github.com/ROCm/MIOpen/pull/2562#discussion_r1436940264